tests/bsim/bluetooth/host/l2cap/frags/src/main.c

```diff
@@ -0,0 +1,477 @@
+/*
+ * The goal of this test is to verify the buf->frags feature works with L2CAP
+ * credit-based channels.
+ *
+ * Copyright (c) 2024 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bstests.h"
+#include "common.h"
+
+#define LOG_MODULE_NAME main
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+CREATE_FLAG(is_connected);
+CREATE_FLAG(flag_l2cap_connected);
+
+#define SMOL_SIZE 10
+#define CHONK_SIZE 50
+#define POOL_NUM 2
+#define PAYLOAD_SIZE (POOL_NUM * (SMOL_SIZE + CHONK_SIZE))
+
+#define L2CAP_MTU PAYLOAD_SIZE
+#define PAYLOAD_NUM 3
+
+NET_BUF_POOL_DEFINE(sdu_rx_pool, 1, BT_L2CAP_SDU_BUF_SIZE(L2CAP_MTU),
+		    CONFIG_BT_CONN_TX_USER_DATA_SIZE, NULL);
+
+NET_BUF_POOL_DEFINE(smol_pool, POOL_NUM, BT_L2CAP_SDU_BUF_SIZE(SMOL_SIZE),
+		    CONFIG_BT_CONN_TX_USER_DATA_SIZE, NULL);
+
+NET_BUF_POOL_DEFINE(chonk_pool, POOL_NUM, CHONK_SIZE,
+		    CONFIG_BT_CONN_TX_USER_DATA_SIZE, NULL);
+
+static uint8_t tx_data[PAYLOAD_SIZE];
+static uint16_t rx_cnt;
+
+K_SEM_DEFINE(sdu_received, 0, 1);
+K_SEM_DEFINE(tx_sem, 1, 1);
+
+struct test_ctx {
+	struct bt_l2cap_le_chan le_chan;
+	size_t tx_left;
```

Comment:
I got confused by this naming, as I was thinking of left / right. 

Consider using
```suggestion
	size_t tx_remaining;
```